addons/mail/models/discuss/ir_websocket.py

```diff
@@ -61,6 +56,11 @@ def _build_presence_channel_list(self, presences):
     @add_guest_to_context
     def _build_bus_channel_list(self, channels):
         channels = list(channels)  # do not alter original list
+        presence_channels = self._build_presence_channel_list(channels)
+        str_presence_channels = {
+            c for c in channels if isinstance(c, str) and c.startswith("odoo-presence-")
+        }
+        channels = [c for c in channels if c not in str_presence_channels]
```

Comment:
You mean removing presence channels from the original `channels` passed to `_build_presence_channel_list`, so altering the parameter? I thought it could lead to some confusion as it is a non-immediately evident side effect but maybe by documenting it we can make it less so